spec/models/mentor/application_spec.rb

```diff
@@ -207,26 +208,181 @@
     end
   end
 
-  describe '#to_param' do
-    it 'returns the underlying active record id' do
-      subject.id = 4711
-      expect(subject.to_param).to eql '4711'
+  describe '#mentor_fav!' do
+    let(:application) { create(:application) }
+
+    subject { m_application.mentor_fav! }
+
+    shared_examples :a_mentor_fav do |choice|
+      let(:other) { (choice % 2) + 1 }
+
+      it 'adds a fav for the chosen project to the persisted application record' do
+        expect { subject }
+          .to change { application.reload.application_data["mentor_fav_project#{choice}"] }
+          .from(nil).to('true')
+      end
+
+      it 'does not change the mentor_fav for the other project' do
+        expect { subject }
+          .not_to change { application.reload.application_data["mentor_fav_project#{other}"] }
+      end
+    end
+
+    context 'when project 1st choice' do
+      let(:m_application) { described_class.new(id: application.id, choice: 1) }
+      include_examples :a_mentor_fav, 1
+    end
+
+    context 'when project 2nd choice' do
+      let(:m_application) { described_class.new(id: application.id, choice: 2) }
+      include_examples :a_mentor_fav, 2
+    end
+  end
+
+  describe '#revoke_mentor_fav!' do
+    let(:application) { create(:application) }
+
+    subject { m_application.revoke_mentor_fav! }
+
+    shared_examples :a_mentor_fav do |choice|
+      let(:other) { (choice % 2) + 1 }
+
+      it 'sets the fav for the chosen project to false' do
+        expect { subject }
+          .to change { application.reload.application_data["mentor_fav_project#{choice}"] }
+          .from(nil).to('false')
+      end
+
+      it 'does not change the mentor_fav for the other project' do
+        expect { subject }
+          .not_to change { application.reload.application_data["mentor_fav_project#{other}"] }
+      end
+    end
+
+    context 'when project 1st choice' do
+      let(:m_application) { described_class.new(id: application.id, choice: 1) }
+      include_examples :a_mentor_fav, 1
+    end
+
+    context 'when project 2nd choice' do
+      let(:m_application) { described_class.new(id: application.id, choice: 2) }
+      include_examples :a_mentor_fav, 2
+    end
+  end
+
+  describe '#sign_off!' do
+    let(:application) { create(:application) }
+    let(:mentor)      { create(:mentor)      }
+
+    def keys_for(choice)
+      ["signed_off_at_project#{choice}", "signed_off_by_project#{choice}"]
+    end
+
+    subject { m_application.sign_off!(as: mentor) }
+
+    shared_examples :a_mentor_sign_off do |choice|
+      let(:other) { (choice % 2) + 1 }
+      let(:now)   { Time.now.utc.to_s }
+
+      before { Timecop.freeze(now) }
+      after { Timecop.return       }
+
+      it 'add the signoff time and user id to the application' do
```

Comment:
Maybe change "add" to "adds" to conform with the "It does something" form, here?